pkg/controller/velerobackup/velerobackup_controller.go

```diff
@@ -130,6 +136,36 @@ func addReconcileRequestLoggerFields(logger log.FieldLogger, request reconcile.R
 	})
 }
 
+func (r *ReconcileBackup) registerHiveObjectWatches(c controller.Controller) error {
+	// Watch for changes to ClusterDeployment
+	err := c.Watch(&source.Kind{Type: &hivev1.ClusterDeployment{}}, &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(r.hiveObjectWatchHandler),
+	})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to SyncSet
+	err = c.Watch(&source.Kind{Type: &hivev1.SyncSet{}}, &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(r.hiveObjectWatchHandler),
+	})
+	if err != nil {
+		return err
+	}
+
+	// If we get here, then no error occurred.
+	return nil
+}
+
+func (r *ReconcileBackup) hiveObjectWatchHandler(a handler.MapObject) []reconcile.Request {
+	// Queue up the NS for this Hive Object
+	retval := []reconcile.Request{
```

Comment:
Since there is so much nesting here, it may be a little easier to read like the following.
```
reqs := make([]reconcile.Request, 1)
reqs[0].Namespace = a.Meta.GetNamespace()
return reqs
```